build.sh

```diff
@@ -29,7 +29,7 @@ function FirstJsonValue {
 }
 
 # If dotnet CLI is installed globally and it matches requested version, use for execution
-if [ -x "$(command -v dotnet)" ] && dotnet --version &>/dev/null; then
+if [ -z "$TEAMCITY_VERSION" ] && [ -x "$(command -v dotnet)" ] && dotnet --version &>/dev/null; then
```

Comment:
Is this saying that on TC we always want it to install dotnet?

Reply:
Yes, I'll add a comment to that effect. Basically it just got to the point where this was the most reliable way of ensuring we were using the correct dotnet sdk.

Reply:
We could make it an explicit env var: `FORCE_DOWNLOAD_OF_DOTNET_AND_USE_THAT_VERSION`. But then we risk someone forgetting to set that.